gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -307,105 +329,137 @@ protected IcebergTable validateGetFilePathsGivenDestState(
   /** @return `paths` after adding to it all paths of every one of `snapshotDefs` */
   protected static Set<Path> withAllSnapshotPaths(Set<Path> paths, MockIcebergTable.SnapshotPaths... snapshotDefs) {
     Arrays.stream(snapshotDefs).flatMap(snapshotDef ->
-        snapshotDef.asSnapshotInfo().getAllPaths().stream()
-    ).forEach(p ->
-        paths.add(new Path(p))
-    );
+            snapshotDef.asSnapshotInfo().getAllPaths().stream())
+        .forEach(p ->
+            paths.add(new Path(p))
+        );
     return paths;
   }
 
   private CopyConfiguration createEmptyCopyConfiguration(FileSystem fs) {
-    return CopyConfiguration.builder(fs, copyConfigProperties)
-        .copyContext(new CopyContext())
-        .build();
+    return CopyConfiguration.builder(fs, copyConfigProperties).copyContext(new CopyContext()).build();
   }
 
   private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
-    List<String> actual = new ArrayList<>();
+    Set<String> actual = new HashSet<>();
```

Comment:
I see this changed... did you find there might be duplicates--and more importantly that our tests ought to ignore rather than call out as a regression?